src/pipeline/src/manager.rs

```diff
@@ -40,8 +43,88 @@ pub mod util;
 /// When it used in cache key, it will be converted to i64 meaning the number of nanoseconds since the epoch.
 pub type PipelineVersion = Option<TimestampNanosecond>;
 
-/// Pipeline info. A tuple of timestamp and pipeline reference.
-pub type PipelineInfo = (Timestamp, PipelineRef);
+/// Unique reference to pipeline name.
+/// A valid pipeline name should be in the format of [<schema>.]<name>
+/// The optional schema is for cross-schema reference.
+/// Note the version can be None while query, which means the latest version of the pipeline.
+#[derive(Debug, Clone)]
+pub struct PipelineName {
+    pub name: String,
+    pub schema: String,
+    pub version: PipelineVersion,
+}
+
+impl PipelineName {
+    /// use [`from_name_and_version`] outside of this crate
+    pub(crate) fn new(name: String, schema: String, version: PipelineVersion) -> Self {
+        Self {
+            name,
+            schema,
+            version,
+        }
+    }
+
+    pub fn from_name_and_version(
+        name: &str,
+        version_str: Option<&str>,
+        default_schema: String,
+    ) -> Result<Self> {
+        let version = to_pipeline_version(version_str)?;
+
+        ensure!(
+            !name.is_empty(),
+            InvalidPipelineNameSnafu {
+                reason: "pipeline name cannot be empty",
+            }
+        );
+
+        let parts = name.split('.').collect::<Vec<&str>>();
```

Comment:
Do we need to consider quoted names such as `"test.db"."test pipeline"`?

Reply:
Looks like we need to process it like 

https://github.com/GreptimeTeam/greptimedb/blob/56e696bd55cf7d708d21b19729801703e5a44a3e/src/session/src/table_name.rs#L43

Use sql parser to parse the name into `ObjectName` at first.